var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -195,7 +182,7 @@ def determine_bootstrap_options(self, spec, withLibs, options):
         options.append("--with-libraries=%s" % ','.join(withLibs))
 
         if '+python' in spec:
-            options.append('--with-python=%s' % python_exe)
+            options.append('--with-python=%s' % spec['python'].executable)
```

Comment:
FYI: `spec['python']` is a `Spec`, which is basically the sub-DAG.  `spec['python'].package` is an instance of `Package`, which is the build system.  If you want a decent way to think about the difference, here you go:

1. Think of the Spec as a descriptor -- you can instantiate a Spec for `foo^bar+baz%intel` without knowing anything about packages  `Foo` or `Bar`.  You cannot concretize or build that Spec without a package called `Foo` and another called `Bar`.
2. The Package does two things: at the class level, it gives you metadata used to concretize Specs (`depends_on()`, `variant()`, etc.).  An instance of `Package` needs to be constructed with a `Spec`, and it provides you with the build system for that package and its dependencies.

We could probably separate (2) out, but IMHO the metaphor of classes-as-package-definitions is a nice one, and it makes it easier for packagers to think about how they can code things up in Python.  We stole that from Homebrew.